log10/secrets.py

```diff
@@ -0,0 +1,40 @@
+"""
+This module provides a class for managing secrets.
+A secret is a key value pair that comes from the enviornment 
+"""
+
+import os
+from dotenv import load_dotenv
+
+
+class SecretsManager:
+    """
+    A class for managing secrets.
+    Override this to work with other secret vaults.
+    """
+
+    def __init__(self, env_file=".env"):
+        self.env: dict[str, str] = {
+            **os.environ,
+            **load_dotenv(env_file),  # override what's in os.environ with .env
```

Comment:
I am not sure what's the appropriate order here (who overrides who) and I don't have strong opinions on either of them.